### DISCUSSION TITLE
Custom dynamic HTML in column

### DISCUSSION THREAD

**Discussion by df999** [filamentphp/filament#6555] [Help]
Is there a simple way to use a `closure` to return some HTML for a column?

Eg. something like:
```PHP
protected function getTableColumns(): array {
    return [
        Tables\Columns\Column::make('name')
                ->formatStateUsing(function ($state, $record) {
                    $color = $record->color;
                    return "<div style='background: #{$color}'>{$record->name}</div>";
                }),
    ];
}
```

One problem with `formatStateUsing` is that I can't put raw HTML in there. Another issue is I'm not sure I can alter the color or classes of the whole cell.

If I use `Tables\Columns\ViewColumn`, then I have to create a whole separate view file just for something simple like this and the app gets bloated. I'm finding I need a lot of simple dynamic styling like this when creating tables.

Is something like this possible?

Thanks!

**Comment by danharrin** [comment]
This is XSS protection. To override it, `return new HtmlString('html here')`

**Comment by df999** [comment]
Okay thanks Dan!
